kiwi/defaults.py

```diff
@@ -139,6 +139,32 @@ def get_platform_name():
             arch = 'ix86'
         return arch
 
+    @staticmethod
+    def get_repository_platform_name(package_manager: str) -> str:
+        """
+        Provides the machine architecture name as used by the repo manager
+
+        This is the architecture name as it is returned by 'uname -m'
+        with exceptions from which we know the uname information needs
+        to be adapted to match with the used repository manager
+
+        :param str package_manager: package manager name
+
+        :return: repo architecture name
+
+        :rtype: str
+        """
+        arch = PLATFORM_MACHINE
+        repo_arch_map = {
+            'zypper': {
+                'armv6l': 'armv6hl',
+                'armv7l': 'armv7hl'
```

Comment:
You may need to add a mapping for `armv8l` (32-bit kernel on 64-bit hardware), but not sure whether to `armv8hl` (don't think we build any special openSUSE packages for it?) or `armv7hl` then.
Not to mention that this may vary by distro - Raspbian/Debian was using `armv5tel` (or similar) on `armv6l`.
Is there some fallback mechanism for either inputs or outputs here? You may also need to consider `i686`/`i586`/etc.

Reply:
> Is there some fallback mechanism for either inputs or outputs here? 

The fallback mechanism is using what comes out of platform.machine() basically ```uname -m```

> Not to mention that this may vary by distro

yes we are aware of it, that's why the map differentiates per repository manager. Zypper is currently the only one for which we define an arch map in cases when platform.machine() is not adequate. Actually Debian distros are the ones were architecture names are almost never match the platform architecture name.

Reply:
We're going to need a similar map for DNF too. Otherwise this method of cross-arch image builds will fail when DNF is selected.

Reply:
> > Is there some fallback mechanism for either inputs or outputs here?
> 
> The fallback mechanism is using what comes out of platform.machine() basically `uname -m`

Not what I meant. I only see a 1:1 uname to RPM arch name mapping here. What if the output doesn't exist?
If on x86 you get `i686` from `uname -m` and either identity-map it or not, only few packages on openSUSE would actually be available for `i686` (glibc maybe) but most would be `i586`, which i686 and later are backwards-compatible with; you can't generally map `i686` to `i586` either though. Similarly, on an `armv8l` system you could fall back to installing `armv7hl` packages, as I pointed out, but would come up empty with either `armv8l` or `armv8hl` for openSUSE RPMs while not forbidden to multi-build them.
RPM maintains mappings of compatible architectures, allowing it to install `i*86` packages as alternative to `x86_64`. So my implied question was, do we need to a) make that map a 1:n list (which could then handle soft-float for Neal, too) and b) have a fallback code path looping over compatible architectures? I.e., what exactly is the mapped target architecture passed to (zypper? rpm? manual filesystem ops?) and does that take fallback to compatible architectures into account?

x32 and ILP32 are more special cases. MIPS has several ABIs, too.

Reply:
I got your point and I also agree with Neal about the arch mapping. Thus I abandoned the idea to map anything as it just causes headaches.

Regarding the multiarch compatibility you mentioned, this is not something kiwi cares for. In kiwi we will create the opportunity to specify one base architecture which is global to the application. On the level of the repository setup it will cause the arch options to be set as available for the package manger in question. In case of zypp.conf it's just "arch: ..." how zypper now handles compatible other archs is outside of kiwi's control